Libraries/LibWeb/HTML/HTMLElement.cpp

```diff
@@ -684,6 +697,8 @@ void HTMLElement::inserted()
 {
     Base::inserted();
     HTMLOrSVGElement::inserted();
+    if (auto* parent_html_element = first_ancestor_of_type<HTMLElement>(); parent_html_element && parent_html_element->is_inert())
+        set_inert(true);
```

Comment:
What about the case where an element is moved from a parent that is inert, to one that is not?

Looking around, I think you need to add inert handling to `HTMLElement::removed_from()` too.